### ISSUE TITLE
[HOLD for payment 2024-06-20] [$1000] IOS- Screen flickering on tapping delete(Transition not smooth)

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#24452]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/24071
## Action Performed:
1. Request money from another user
2. Tap the report preview (this will navigate you to the iou report)
3. Hover the IOU preview and tap the thrash can icon
4. Verify that a modal confirming if you want to delete the request shows up
5. Tap Cancel and verify that the modal closes
6. Repeat step 1-4 but this time tap Delete
7. Verify that the request is deleted
8. Request money again
9. Tap the report preview (this will navigate you to the iou report)
10. Tap the IOU preview (this will navigate you to the transaction thread report)
11. Tap the three dot menu at the top right of the page and then Delete request
12. Verify that the same confirmation modal shows up
13. Tap Cancel and verify that the modal closes
14. Tap the three dot menu at the top right of the page and then Delete request again

## Expected Result:
Delete request confirmation modal displayed without any issues

## Actual Result:
Screen flickering on tapping delete(Transition not smooth)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.53.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/8434d9ad-4eec-414b-9a8a-3154e175e045

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019298424ebc35a6e6</li>
        <li>Upwork Job ID: 1696029632941596672</li>
        <li>Last Price Increase: 2023-10-02</li>
<li>Automatic offers: </li>
<ul>
<li>0xmiroslav | Reviewer | 27008615</li>
<li>ntdiary | Contributor | 27008616</li>
<li>getusha | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @kadiealexander</details>

**Comment by sophiepintoraetz** [comment]
Sounds good - closing this for now.

**Comment by sophiepintoraetz** [comment]
I need to find the time to reproduce this!

**Comment by sophiepintoraetz** [comment]
Today!

**Comment by sophiepintoraetz** [comment]
Yup - I'm able to reproduce

https://github.com/Expensify/App/assets/42940078/3e2806bc-2304-498c-93bd-c1b0e101f727

**Comment by kubabutkiewicz** [comment]
Hi, I'm Jakub from Callstack - expert contributor group - and I would like to take a look at this issue.

**Comment by kubabutkiewicz** [comment]
Hi @sophiepintoraetz , could you give more details about the device and OS version which you are using ?

**Comment by sophiepintoraetz** [comment]
Sure - iphone 14, iOS 16.6 app version v1.3.58-3

**Comment by kubabutkiewicz** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Screen flickering on tapping delete request (Transition not smooth)
### What is the root cause of that problem?
The problem here is that when we are trying to open modal by clicking Delete Request and in the same time keyboard is open (we have an input focused) the modal opening but in the same time keyboard is still open , because event for closing keyboard is not firing, so keyboard is just hidden behind modal which is causing flickering.
### What changes do you think we should make in order to solve the problem?
We have state of modal in onyx so we can detect when modal is going to open/close so in ReportActionCompose we can listen to changes of modal state and blur or focus the Input to hide or show keyboard. That will do that animation of opening/closing modal will be smooth.

### What alternative solutions did you explore? (Optional)

**Comment by sophiepintoraetz** [comment]
Just waiting on @0xmiroslav to review proposals

**Comment by sophiepintoraetz** [comment]
[Waiting](https://expensify.slack.com/archives/C02NK2DQWUX/p1693871063968319) on @0xmiroslav

**Comment by sophiepintoraetz** [comment]
Bumped again.

**Comment by kubabutkiewicz** [comment]
@0xmiroslav can you explain what you mean by permal link?

**Comment by kubabutkiewicz** [comment]
And also can you explain why my proposal is not complete ? @0xmiroslav 
> Note: we should keep composer focused state before opening modal.
I think in that case we need to do that because flickering is cause by opened keyboard.
So we firstly need to hide keyboard which we can do by dismissing (which we cause unfocus of composer) and then when we will focus it again if we close the modal.

**Comment by sophiepintoraetz** [comment]
Waiting for a review from @0xmiroslav

**Comment by sophiepintoraetz** [comment]
@0xmiroslav - waiting for you.

**Comment by sophiepintoraetz** [comment]
@0xmiroslav - can you please work with the existing proposals? @ntdiary is a C+ and @kubabutkiewicz is from one of our partner agencies so their proposals won't be too far off the mark (@kubabutkiewicz - if you don't mind resubmitting your proposal to include the permalink?)

**Comment by sophiepintoraetz** [comment]
[Bumped in slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1695355162424939).

**Comment by sophiepintoraetz** [comment]
Tagged the right contributor this time - will look at doubling this by the end of the week.

**Comment by kubabutkiewicz** [comment]
@0xmiroslav The proposal from @ntdiary looks good to me, usage of refs here looks reasonable here because it will not cause additional re-renders of the components in comparison of using onyx store. I also think it will be hard to find solution which will remove this issue once for all modals. But we can create maybe an issue to followup on that to try to find that solution.

**Comment by sophiepintoraetz** [comment]
@0xmiroslav - thoughts here?

**Comment by sophiepintoraetz** [comment]
Waiting on @ntdiary to re-submit their proposal.

**Comment by danieldoglas** [comment]
proposal looks good to me, assigning @ntdiary

**Comment by sophiepintoraetz** [comment]
Hi @joekaufmanexpensify - just needing a baby sitter for this issue in case it's resolved before I'm back in office 17 Oct. All that should be needed from your is a payment to the contributor and C+ (amounts TBD).

**Comment by joekaufmanexpensify** [comment]
Pending PR

**Comment by danieldoglas** [comment]
@joekaufmanexpensify I'll be OOO for the next 2 weeks, can you please assign the engineering flag again on this one? Thanks!

**Comment by joekaufmanexpensify** [comment]
Assigning another engineer per above comment

**Comment by joekaufmanexpensify** [comment]
@ntdiary will the PR here be up soon?

**Comment by joekaufmanexpensify** [comment]
Got it, sounds good!

**Comment by sophiepintoraetz** [comment]
Thank you, @joekaufmanexpensify!

**Comment by sophiepintoraetz** [comment]
I think we're still on hold here, effectively.

**Comment by sophiepintoraetz** [comment]
Hmm, not overdue Melvin!

**Comment by sophiepintoraetz** [comment]
@ntdiary do you mind linking the PR here? Tried to scroll back through the history but couldn't find it.

**Comment by sophiepintoraetz** [comment]
I imagine we'll get traction once we gear up from the holidays.

**Comment by puneetlath** [comment]
It's not super clear to me what the deal is with this issue, but I don't think I'm needed here so I'm going to drop off. Feel free to re-assign me if I'm mistaken.

**Comment by sophiepintoraetz** [comment]
Not overdue - PR is still in review.

**Comment by mallenexpensify** [comment]
@ntdiary since you're a C+, I'm not assigning another now.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Julesssss** [comment]
This should be fixed along with [this issue](https://github.com/Expensify/App/issues/29011)

**Comment by kadiealexander** [comment]
For clarity:

Issue assignee: @ntdiary
C+: @parasharrajat

**Comment by kadiealexander** [comment]
@ntdiary any updates here?

**Comment by kadiealexander** [comment]
@parasharrajat any update on your PR review?

**Comment by kadiealexander** [comment]
Bumped Rajat [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1707967551220519)

**Comment by kadiealexander** [comment]
Reassigning for urgency, thanks @getusha!

**Comment by kadiealexander** [comment]
Not overdue!

**Comment by getusha** [comment]
PR in review.

**Comment by Julesssss** [comment]
PR review still in progress

**Comment by kadiealexander** [comment]
Not overdue

**Comment by Julesssss** [comment]
Slow and steady progress being made, I request another main merge to keep the TS changes up to date.

**Comment by Julesssss** [comment]
The linked PR has been merged.

**Comment by Julesssss** [comment]
Most recent update is [here](https://github.com/Expensify/App/pull/29199), the recent architecture change requires an update to the previous solution.

**Comment by Julesssss** [comment]
Hey @ntdiary, any update on the above plan?

**Comment by Julesssss** [comment]
Great, the [PR](https://github.com/Expensify/App/pull/42423) tested well and was just merged 👍

**Comment by Julesssss** [comment]
Let's keep an eye out for other flickering issues

**Comment by sakluger** [comment]
Hey folks, was this issue split up into multiple issues/PRs? We have been holding https://github.com/Expensify/App/issues/31224 on this issue and I wanted to check if this was complete before opening that other issue back up to new proposals.

**Comment by Julesssss** [comment]
@sakluger this was related to two other other PRs I believe, but this should be the last of them 👍

**Comment by Julesssss** [comment]
Ah thanks, I forgot about the later phases.

**Comment by kadiealexander** [comment]
@getusha please complete the checklist!

**Comment by kadiealexander** [comment]
Payouts due:

- [ ] Contributor: $1000 @ntdiary - https://www.upwork.com/nx/wm/offer/27008616
- [ ] Contributor+: $1000 @getusha - https://www.upwork.com/nx/wm/offer/27008615

**Comment by Julesssss** [comment]
Hey @kadiealexander is this completed?

**Comment by kadiealexander** [comment]
@getusha still needs to complete the BugZero checklist so I can pay out.

**Comment by Julesssss** [comment]
Waiting on @getusha

**Comment by getusha** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@getusha] The PR that introduced the bug has been identified. Link to the PR: **N/a** This is an edge case and i don't think there is a specific PR that's responsible for this issue
 [@getusha] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **N/a**
 [@getusha] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/a**
 [@getusha] Determine if we should create a regression test for this bug. **Yes**
 [@getusha] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Submit an expense to another user
2. Press the expense preview to open the request details page
3. Focus the main composer input
4. Press the three dot menu at the top right of the page
5. Press the Delete expense
6. Verify the confirmation modal is opened smoothly and the keyboard didn't flicker

Do we agree 👍 or 👎

**Comment by Julesssss** [comment]
Sorry @getusha I should have said earlier, but I don't think we need a specific regression test for this flickering.

**Comment by getusha** [comment]
> Sorry @getusha I should have said earlier, but I don't think we need a specific regression test for this flickering.

we can skip it since we had this issue happening several places. my reasoning was to just have one simpler test case to ensure it gets caught if any changes are made to the new focus manager.

**Comment by getusha** [comment]
@kadiealexander friendly bump for payment on this one, i am not able to find the offer :)

**Comment by kadiealexander** [comment]
Hey @getusha is this offer visible for you? https://www.upwork.com/nx/wm/offer/103274633

**Comment by kadiealexander** [comment]
@getusha bump!

**Comment by getusha** [comment]
@kadiealexander accepted the offer, thank you!
